blocksubscriptions.go

```diff
@@ -35,33 +40,50 @@ type blockSubscription struct {
 
 // sendSubscribedMsg sends all block subscribers a message if they request this
 // type.
+//
 // TODO(aakselrod): Refactor so we're able to handle more message types in new
 // package.
 func (s *ChainService) sendSubscribedMsg(bm *blockMessage) {
-	var subChan chan<- wire.BlockHeader
+
 	s.mtxSubscribers.RLock()
 	for sub := range s.blockSubscribers {
-		switch bm.msgType {
-		case connectBasic:
-			subChan = sub.onConnectBasic
-		case disconnect:
-			subChan = sub.onDisconnect
-		default:
-			// TODO: Return a useful error when factored out into
-			// its own package.
-			panic("invalid message type")
-		}
-		if subChan != nil {
-			select {
-			case sub.notifyBlock <- bm:
-			case <-sub.quit:
-			case <-sub.intQuit:
-			}
-		}
+		sendMsgToSubscriber(sub, bm)
 	}
 	s.mtxSubscribers.RUnlock()
 }
 
+// sendMsgToSubscriber is a helper function that sends the target message to
+// the subscription client over the proper channel based on the type of the new
```

Comment:
godoc says it "sends [..] over the proper channel", but it is always sent on the `notifyBlocks` channel.  

Actually looks like the `selectChan` method that handles this block message later will handle the case where the sub channel is `nil`, so this confusing method could be simplified significantly. no blocker for this pr though, as this is equivalent to what was already here.